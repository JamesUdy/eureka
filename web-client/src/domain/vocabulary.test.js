require('chai')
  .should();

function tokenize(field) {
  let parts = field.split(' ');
  let type = parts[0];
  let identifier = parts[1];
  return {
    type: type,
    words: splitIdentifier(identifier),
  };
}

function splitIdentifier(identifier) {
  if (identifier.indexOf('_') !== -1) {
    return identifier.split('_');
  }
  return identifier.split(/(?=[A-Z])/);
}

describe('tokenization', () => {
  it('should tokenize a simple field', () => {
    const tokens = tokenize('int x');
    tokens.should.deep.equal({
      type: 'int',
      words: ['x'],
    });
  });

  it('should tokenize a field in camel case', function () {
    const actual = tokenize('int xCamelCase');
    actual.should.deep.equal({
      type: 'int',
      words: ['x', 'Camel', 'Case'],
    });
  });

  it('should tokenize a field in snake case', function () {
    const actual = tokenize('int x_snake_case');
    actual.should.deep.equal({
      type: 'int',
      words: ['x', 'snake', 'case']
    });
  });

  it('should tokenize constants', function () {
    const actual = tokenize('int X_CONSTANT');
    actual.should.deep.equal({
      type: 'int',
      words: ['X', 'CONSTANT']
    });
  });

  it('should not split types into individual words', function () {
    const actual = tokenize('StoryBook storyBook');
    actual.should.deep.equal({
      type: 'StoryBook',
      words: ['story', 'Book'],
    });
  });

  it('should split identifiers that start with a capital letter into individual words', function () {
    let actual = tokenize('String QuestionTitle');
    actual.should.deep.equal({
      type: 'String',
      words: ['Question', 'Title'],
    });
  });
});
