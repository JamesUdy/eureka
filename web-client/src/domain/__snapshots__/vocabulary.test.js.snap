// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class vocabulary should build a member vocabulary from a given graph 1`] = `
Object {
  "types": Array [
    "String",
    "int",
    "String",
    "void",
    "void",
    "String",
    "void",
    "int",
    "String",
    "String",
  ],
  "words": Array [
    "increment",
    "Version",
    "random",
    "what",
    "Version",
    "say",
    "Goodbye",
    "say",
    "Hello",
    "get",
    "Name",
    "version",
    "Code",
    "version",
    "name",
  ],
}
`;

exports[`class vocabulary should build vocabulary for state 1`] = `
Object {
  "types": Object {
    "String": 5,
    "int": 2,
    "void": 3,
  },
  "words": Object {
    "code": 1,
    "get": 1,
    "goodbye": 1,
    "hello": 1,
    "increment": 1,
    "name": 2,
    "random": 1,
    "say": 2,
    "version": 4,
    "what": 1,
  },
}
`;

exports[`class vocabulary should build vocabulary stats from the vocabulary object 1`] = `
Object {
  "types": Object {
    "String": 5,
    "int": 2,
    "void": 3,
  },
  "words": Object {
    "code": 1,
    "get": 1,
    "goodbye": 1,
    "hello": 1,
    "increment": 1,
    "name": 2,
    "random": 1,
    "say": 2,
    "version": 4,
    "what": 1,
  },
}
`;
